core/src/commonMain/kotlin/com/ustadmobile/core/domain/report/model/ReportFilter2.kt

```diff
@@ -1,57 +1,81 @@
-
-    var reportFilterSeriesUid: Int = 0,
-
-    var reportFilterField: Int = FIELD_PERSON_GENDER,
-
-    var reportFilterCondition: Int = 0,
-
-    var reportFilterValue: String? = null,
-
-    var reportFilterDropDownValue: Int = 0,
-
-    var reportFilterValueBetweenX: String? = null,
-
-    var reportFilterValueBetweenY: String? = null,
-) {
+/** Enum representing different comparison types */
+enum class Comparisons(val value: Int) {
+    EQUALS(1),
+    NOT_EQUALS(2),
+    GREATER(3),
+    LESSER(4),
+    BETWEEN(5);
 
     companion object {
+        // Map enum values to their corresponding string resources
+        private val stringResourceMap = mapOf(
+            EQUALS to MR.strings.equals,
+            NOT_EQUALS to MR.strings.not_equals,
+            GREATER to MR.strings.greater,
+            LESSER to MR.strings.lesser,
+            BETWEEN to MR.strings.between
+        )
+
+        // Function to get the string resource for a given enum value
+        fun getStringResourceForComparison(comparison: Comparisons): StringResource {
+            return stringResourceMap[comparison] ?: MR.strings.equals
+        }
+    }
+}
 
-        const val FIELD_PERSON_GENDER = 100
-
-        const val FIELD_PERSON_AGE = 101
-
-        const val FIELD_CONTENT_COMPLETION = 102
-
-        const val FIELD_CONTENT_ENTRY = 103
-
-        const val FIELD_CONTENT_PROGRESS = 104
-
-        const val FIELD_ATTENDANCE_PERCENTAGE = 105
-
-        const val FIELD_CLAZZ_ENROLMENT_OUTCOME = 106
-
-        const val FIELD_CLAZZ_ENROLMENT_LEAVING_REASON = 107
-
-        const val CONDITION_IS = 200
 
-        const val CONDITION_IS_NOT = 201
+/** Sealed class representing different types of report filters */
+@Serializable
+sealed class ReportFilter2(
+    val comparisonTypes: List<Comparisons>,
+) {
+    @Serializable
+     class GenderFilter : ReportFilter2(
+        comparisonTypes = listOf(Comparisons.EQUALS, Comparisons.NOT_EQUALS)
+    )
+
+    @Serializable
+     class AgeFilter : ReportFilter2(
+        comparisonTypes = listOf(
+            Comparisons.EQUALS,
+            Comparisons.NOT_EQUALS,
+            Comparisons.GREATER,
+            Comparisons.LESSER,
+            Comparisons.BETWEEN
+        )
+    )
+
+    @Serializable
+     class ContentCompletionFilter: ReportFilter2(
+        comparisonTypes = listOf(Comparisons.EQUALS, Comparisons.GREATER, Comparisons.LESSER)
+    )
+}
 
-        const val CONDITION_GREATER_THAN = 202
+@Serializable
+data class ReportFilter3(
+    var reportFilterUid: Int = 0,
 
-        const val CONDITION_LESS_THAN = 203
+    var reportFilterSeriesUid: Int = 0,
 
-        const val CONDITION_BETWEEN = 205
+    var reportFilterField: FilterType? = null,
 
-        const val CONDITION_IN_LIST = 206
+    var reportFilterCondition: Comparisons? = null,
 
-        const val CONDITION_NOT_IN_LIST = 207
+    var reportFilterValue: String? = ""
+)
 
+fun getComparisonSymbol(comparison: Comparisons): String {
```

Comment:
Again - there is no reason to add this extra function. The comparison symbol can just be an extra field on the enum class.